### PR TITLE
refine description of  txn-total-size-limit and tidb_mem_quota_query

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -2101,9 +2101,10 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 默认值：`1073741824` (1 GiB)
 - 范围：`[-1, 9223372036854775807]`
 - 单位：字节
-- 这个变量用来设置一条查询语句的内存使用阈值。
-- 如果一条查询语句执行过程中使用的内存空间超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。
-- 在 v6.1.0 之前这个开关通过 TiDB 配置文件 (`mem-quota-query`) 进行配置，且作用域为 `SESSION`。升级到 v6.1.0 时会自动继承原有设置，作用域变更为 `SESSION | GLOBAL`。
+- 在 v6.1.0 之前的版本中，作用域为 `SESSION`。v6.1.0 及之后的版本时，作用域变更为 `SESSION | GLOBAL`。
+- 在 v6.5.0 之前的版本中，该变量用来设置单条查询的内存使用阈值，如果单条查询执行过程中使用的内存量超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。
+- 在 v6.5.0 及之后的版本中，该变量用来设置单个会话的内存使用阈值，如果单个会话执行过程中使用的内存量超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。值得注意的是，自 v6.5.0 版本开始，单个事务的内存使用也会被累计入所在会话的内存使用量，v6.5.0 及之后版本对事务内存的控制行为详见 [txn-total-size-limit](/tidb-configuration-file.md#txn-total-size-limit)。
```
Comment (hfxsd) [inline system-variables.md:2106]:
```suggestion
- 在 v6.5.0 及之后的版本中，该变量用来设置单个会话的内存使用阈值，如果单个会话执行过程中使用的内存量超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。值得注意的是，自 v6.5.0 版本开始，单个事务的内存使用也会被累计计入所在会话的内存使用量，v6.5.0 及之后版本对事务内存的控制行为详见 [txn-total-size-limit](/tidb-configuration-file.md#txn-total-size-limit)。
```

File: tidb-configuration-file.md [hunk 2]
```diff
@@ -403,6 +403,9 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：104857600
 + 单位：Byte
 + 单个事务中，所有 key-value 记录的总大小不能超过该限制。该配置项的最大值不超过 `1099511627776`（表示 1TB）。注意，如果使用了以 `Kafka` 为下游消费者的 `binlog`，如：`arbiter` 集群，该配置项的值不能超过 `1073741824`（表示 1GB），因为这是 `Kafka` 的处理单条消息的最大限制，超过该限制 `Kafka` 将会报错。
++ 在 v6.5.0 及之后的版本中，不再推荐使用该配置项，事务的内存大小会被累计入所在会话的内存使用量中，并由 [tidb_mem_quota_query](/system-variables.md#tidb_mem_quota_query) 变量在单个会话内存超阈值时采取控制行为。为了向前兼容，由低版本升级至 v6.5.0 及更高版本时，该配置项的行为如下所述:
```
Comment (hfxsd) [inline tidb-configuration-file.md:406]:
```suggestion
+ 在 v6.5.0 及之后的版本中，不再推荐使用该配置项，事务的内存大小会被累计计入所在会话的内存使用量中，并由 [tidb_mem_quota_query](/system-variables.md#tidb_mem_quota_query) 变量在单个会话内存超阈值时采取控制行为。为了向前兼容，由低版本升级至 v6.5.0 及更高版本时，该配置项的行为如下所述:
```

File: tidb-configuration-file.md [hunk 3]
```diff
@@ -403,6 +403,9 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
+	+ 若该配置项设置值不等于默认值（104857600），升级前后该配置项对单个事务大小的限制行为不会发生变化，该变量仍会生效，事务内存大小不由 tidb_mem_quota_query 控制。
+	+ 若该配置未被设置，或设置值等于默认值（104857600），升级后事务内存大小将会累计入所在会话的内存使用量中，由[tidb_mem_quota_query 变量控制。
```
Comment (hfxsd) [inline tidb-configuration-file.md:408]:
```suggestion
	+ 若该配置未被设置，或设置值等于默认值（104857600），升级后事务内存大小将会累计入所在会话的内存使用量中，由[tidb_mem_quota_query 变量控制。
	+ 若该配置项设置值不等于默认值（104857600），升级前后该配置项对单个事务大小的限制行为不会发生变化，该变量仍会生效，事务内存大小不由 tidb_mem_quota_query 控制。
```

File: tidb-configuration-file.md [hunk 4]
```diff
@@ -403,6 +403,9 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：104857600
 + 单位：Byte
 + 单个事务中，所有 key-value 记录的总大小不能超过该限制。该配置项的最大值不超过 `1099511627776`（表示 1TB）。注意，如果使用了以 `Kafka` 为下游消费者的 `binlog`，如：`arbiter` 集群，该配置项的值不能超过 `1073741824`（表示 1GB），因为这是 `Kafka` 的处理单条消息的最大限制，超过该限制 `Kafka` 将会报错。
++ 在 v6.5.0 及之后的版本中，不再推荐使用该配置项，事务的内存大小会被累计计入所在会话的内存使用量中，并由 [tidb_mem_quota_query](/system-variables.md#tidb_mem_quota_query) 变量在单个会话内存超阈值时采取控制行为。为了向前兼容，由低版本升级至 v6.5.0 及更高版本时，该配置项的行为如下所述:
+    + 若该配置未被设置，或设置值等于默认值（104857600），升级后事务内存大小将会累计入所在会话的内存使用量中，由[tidb_mem_quota_query 变量控制。
```
Comment (XuHuaiyu) [inline tidb-configuration-file.md:407]:
```suggestion
    + 若该配置未被设置，或设置值等于默认值（104857600），升级后事务内存大小将会累计入所在会话的内存使用量中，由 tidb_mem_quota_query 变量控制。
```

File: tidb-configuration-file.md [hunk 5]
```diff
@@ -403,6 +403,9 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：104857600
 + 单位：Byte
 + 单个事务中，所有 key-value 记录的总大小不能超过该限制。该配置项的最大值不超过 `1099511627776`（表示 1TB）。注意，如果使用了以 `Kafka` 为下游消费者的 `binlog`，如：`arbiter` 集群，该配置项的值不能超过 `1073741824`（表示 1GB），因为这是 `Kafka` 的处理单条消息的最大限制，超过该限制 `Kafka` 将会报错。
++ 在 v6.5.0 及之后的版本中，不再推荐使用该配置项，事务的内存大小会被累计计入所在会话的内存使用量中，并由 [tidb_mem_quota_query](/system-variables.md#tidb_mem_quota_query) 变量在单个会话内存超阈值时采取控制行为。为了向前兼容，由低版本升级至 v6.5.0 及更高版本时，该配置项的行为如下所述:
+    + 若该配置未被设置，或设置值等于默认值（104857600），升级后事务内存大小将会累计入所在会话的内存使用量中，由 tidb_mem_quota_query 变量控制。
+    + 若该配置项设置值不等于默认值（104857600），升级前后该配置项对单个事务大小的限制行为不会发生变化，该变量仍会生效，事务内存大小不由 tidb_mem_quota_query 控制。
```
Comment (hfxsd) [inline tidb-configuration-file.md:408]:
```suggestion
+ 在 v6.5.0 及之后的版本中，不再推荐使用该配置项，事务的内存大小会被累计计入所在会话的内存使用量中，并由 [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query) 变量在单个会话内存超阈值时采取控制行为。为了向前兼容，由低版本升级至 v6.5.0 及更高版本时，该配置项的行为如下所述:
    + 若该配置未被设置，或设置值等于默认值（104857600），升级后事务内存大小将会累计入所在会话的内存使用量中，由 `tidb_mem_quota_query` 变量控制。
    + 若该配置项设置值不等于默认值（104857600），升级前后该配置项对单个事务大小的限制行为不会发生变化，该变量仍会生效，事务内存大小不由 `tidb_mem_quota_query` 控制。
```

File: system-variables.md [hunk 6]
```diff
@@ -2101,9 +2101,10 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 默认值：`1073741824` (1 GiB)
 - 范围：`[-1, 9223372036854775807]`
 - 单位：字节
-- 这个变量用来设置一条查询语句的内存使用阈值。
-- 如果一条查询语句执行过程中使用的内存空间超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。
-- 在 v6.1.0 之前这个开关通过 TiDB 配置文件 (`mem-quota-query`) 进行配置，且作用域为 `SESSION`。升级到 v6.1.0 时会自动继承原有设置，作用域变更为 `SESSION | GLOBAL`。
+- 在 v6.1.0 之前的版本中，作用域为 `SESSION`。v6.1.0 及之后的版本时，作用域变更为 `SESSION | GLOBAL`。
+- 在 v6.5.0 之前的版本中，该变量用来设置单条查询的内存使用阈值，如果单条查询执行过程中使用的内存量超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。
+- 在 v6.5.0 及之后的版本中，该变量用来设置单个会话的内存使用阈值，如果单个会话执行过程中使用的内存量超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。值得注意的是，自 v6.5.0 版本开始，单个事务的内存使用也会被累计计入所在会话的内存使用量，v6.5.0 及之后版本对事务内存的控制行为详见 [txn-total-size-limit](/tidb-configuration-file.md#txn-total-size-limit)。
```
Comment (shichun-0415) [inline system-variables.md:2106]:
```suggestion
- 在 v6.5.0 及之后的版本中，该变量用来设置单个会话的内存使用阈值，如果单个会话执行过程中使用的内存量超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。注意，自 v6.5.0 版本开始，单个事务的内存使用也会被累计计入所在会话的内存使用量。v6.5.0 及之后版本对事务内存的控制行为，详见 [txn-total-size-limit](/tidb-configuration-file.md#txn-total-size-limit)。
```

File: system-variables.md [hunk 7]
```diff
@@ -2101,9 +2101,10 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 默认值：`1073741824` (1 GiB)
 - 范围：`[-1, 9223372036854775807]`
 - 单位：字节
-- 这个变量用来设置一条查询语句的内存使用阈值。
-- 如果一条查询语句执行过程中使用的内存空间超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。
-- 在 v6.1.0 之前这个开关通过 TiDB 配置文件 (`mem-quota-query`) 进行配置，且作用域为 `SESSION`。升级到 v6.1.0 时会自动继承原有设置，作用域变更为 `SESSION | GLOBAL`。
+- 在 v6.1.0 之前的版本中，作用域为 `SESSION`。v6.1.0 及之后的版本时，作用域变更为 `SESSION | GLOBAL`。
```
Comment (shichun-0415) [inline system-variables.md:2104]:
```suggestion
- 在 v6.1.0 之前的版本中，作用域为 `SESSION`。v6.1.0 及之后的版本，作用域变更为 `SESSION | GLOBAL`。
```

File: system-variables.md [hunk 8]
```diff
@@ -2101,9 +2101,10 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+- 在 v6.5.0 之前的版本中，该变量用来设置单条查询的内存使用阈值，如果单条查询执行过程中使用的内存量超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。
+- 在 v6.5.0 及之后的版本中，该变量用来设置单个会话的内存使用阈值，如果单个会话执行过程中使用的内存量超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。值得注意的是，自 v6.5.0 版本开始，单个事务的内存使用也会被累计计入所在会话的内存使用量，v6.5.0 及之后版本对事务内存的控制行为详见 [txn-total-size-limit](/tidb-configuration-file.md#txn-total-size-limit)。
+- 当变量值为 0 或 -1 时，表示内存阈值为正无穷。此外，当变量值设置值小于 128 时，将默认被设置为 128。
```
Comment (shichun-0415) [inline system-variables.md:2107]:
```suggestion
- 当变量值为 `0` 或 `-1` 时，表示内存阈值为正无穷。此外，当变量值小于 `128` 时，将默认被设置为 `128`。
```

File: tidb-configuration-file.md [hunk 9]
```diff
@@ -403,6 +403,9 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
 + 默认值：104857600
 + 单位：Byte
 + 单个事务中，所有 key-value 记录的总大小不能超过该限制。该配置项的最大值不超过 `1099511627776`（表示 1TB）。注意，如果使用了以 `Kafka` 为下游消费者的 `binlog`，如：`arbiter` 集群，该配置项的值不能超过 `1073741824`（表示 1GB），因为这是 `Kafka` 的处理单条消息的最大限制，超过该限制 `Kafka` 将会报错。
++ 在 v6.5.0 及之后的版本中，不再推荐使用该配置项，事务的内存大小会被累计计入所在会话的内存使用量中，并由 [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query) 变量在单个会话内存超阈值时采取控制行为。为了向前兼容，由低版本升级至 v6.5.0 及更高版本时，该配置项的行为如下所述:
+    + 若该配置未被设置，或设置值等于默认值（104857600），升级后事务内存大小将会累计入所在会话的内存使用量中，由 `tidb_mem_quota_query` 变量控制。
```
Comment (shichun-0415) [inline tidb-configuration-file.md:407]:
```suggestion
    + 若该配置项未设置，或设置为默认值 (`104857600`)，升级后事务内存大小将会计入所在会话的内存使用中，由 `tidb_mem_quota_query` 变量控制。
```

File: tidb-configuration-file.md [hunk 10]
```diff
@@ -403,6 +403,9 @@ TiDB 配置文件比命令行参数支持更多的选项。你可以在 [config/
...
+    + 若该配置项设置值不等于默认值（104857600），升级前后该配置项对单个事务大小的限制行为不会发生变化，该变量仍会生效，事务内存大小不由 `tidb_mem_quota_query` 控制。
```
Comment (shichun-0415) [inline tidb-configuration-file.md:408]:
```suggestion
    + 若该配置项未设为默认值 (`104857600`)，升级前后该配置项仍生效，对单个事务大小的限制行为不会发生变化，事务内存大小不由 `tidb_mem_quota_query` 控制。
```

File: system-variables.md [hunk 11]
```diff
@@ -2101,9 +2101,10 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 默认值：`1073741824` (1 GiB)
 - 范围：`[-1, 9223372036854775807]`
 - 单位：字节
-- 这个变量用来设置一条查询语句的内存使用阈值。
-- 如果一条查询语句执行过程中使用的内存空间超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。
-- 在 v6.1.0 之前这个开关通过 TiDB 配置文件 (`mem-quota-query`) 进行配置，且作用域为 `SESSION`。升级到 v6.1.0 时会自动继承原有设置，作用域变更为 `SESSION | GLOBAL`。
+- 在 v6.1.0 之前的版本中，作用域为 `SESSION`。v6.1.0 及之后的版本时，作用域变更为 `SESSION | GLOBAL`。
+- 在 v6.5.0 之前的版本中，该变量用来设置单条查询的内存使用阈值，如果单条查询执行过程中使用的内存量超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。
```
Comment (songrijie) [inline system-variables.md:2105]:
```suggestion
- 在 v6.5.0 之前的版本中，该变量用来设置单条查询的内存使用限制，如果单条查询执行过程中使用的内存量超过该阈值，会触发系统变量 [`tidb_mem_oom_action`](#tidb_mem_oom_action-从-v610-版本开始引入) 中指定的行为。
```

File: tidb-configuration-file.md (hunk 10 above)
Comment (songrijie) [reply tidb-configuration-file.md:408]: 最后加一句观点，“如果客户在 v6.5.0 之前的版本设置了 `txn-total-size-limit` 用来控制事务大小，建议移除这个配置项，转而设置合理的 [`tidb_mem_quota_query`](/system-variables.md#tidb_mem_quota_query) 来提升内存使用效率。 “

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Review by hfxsd — APPROVED** [review]

**Pull request merged**